### PR TITLE
Collapse Maneuvers (scenario2)

### PR DESCRIPTION
**Pull request by kdiluca**
# Issue

This PR fixes an issue in guidance where we add an extra maneuver. This PR combines a current maneuver short length non-internal edges (left or right) with next maneuver that is a kRampStraight.

This will fix number 2 in mapbox/nav-api-intake#96

## Tasklist

 - [X] Add tests
 - [X] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [ ] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by kdiluca** [comment]
Only expected diffs:

image

image
image
image
image

File: src/odin/maneuversbuilder.cc [hunk 1]
```diff
@@ -519,6 +521,32 @@ void ManeuversBuilder::Combine(std::list<Maneuver>& maneuvers) {
...
+        if (is_first_man) {
+          next_man->set_type(DirectionsLeg_Maneuver_Type_kStart);
+        } else {
+          // Set maneuver type to 'none' so the type will be processed again
+          next_man->set_type(DirectionsLeg_Maneuver_Type_kNone);
+        }
+        curr_man = CombineUnspecifiedInternalManeuver(maneuvers, prev_man, curr_man, next_man,
+                                                      next_man->type());
+
+        if (is_first_man) {
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:545]: remove this too

File: src/odin/maneuversbuilder.cc [hunk 2]
```diff
@@ -519,6 +521,32 @@ void ManeuversBuilder::Combine(std::list<Maneuver>& maneuvers) {
...
+                (ManeuversBuilder::DetermineRelativeDirection(curr_man->turn_degree()) ==
+                 Maneuver::RelativeDirection::kRight)) &&
+               next_man->type() == DirectionsLeg_Maneuver_Type_kRampStraight) {
+        LOG_TRACE(
+            "+++ Combine: current non-internal turn that is very short in length with the next straight ramp maneuver +++");
+        if (is_first_man) {
+          next_man->set_type(DirectionsLeg_Maneuver_Type_kStart);
+        } else {
+          // Set maneuver type to 'none' so the type will be processed again
+          next_man->set_type(DirectionsLeg_Maneuver_Type_kNone);
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:540]: dont need this - just pass DirectionsLeg_Maneuver_Type_kNone

File: test/gurka/test_instructions_collapse_double_turn.cc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                                                DirectionsLeg_Maneuver_Type_kLeft,
+                                                DirectionsLeg_Maneuver_Type_kRampLeft,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+
+  // Verify maneuver begin path indexes
+  gurka::assert::raw::expect_maneuver_begin_path_indexes(result, {0, 1, 2, 5});
+
+  int maneuver_index = 1;
+
+  // Verify the L U-turn instructions
```
Comment (dgearhart) [inline test/gurka/test_instructions_collapse_double_turn.cc:65]: not u-turn

File: test/gurka/test_instructions_collapse_double_turn.cc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  gurka::assert::raw::expect_instructions_at_maneuver_index(
+      result, maneuver_index, "Turn left onto South River Road.", "Turn left onto South River Road.",
+      "Turn left onto South River Road. Then Turn left to take the ramp.", "Continue for 50 meters.");
+
+  maneuver_index = 2;
+
+  // Verify the L U-turn instructions
```
Comment (dgearhart) [inline test/gurka/test_instructions_collapse_double_turn.cc:72]: not u-turn

**Comment by dgearhart** [comment]
Tests are looking good - I am running one more set
@kdiluca remember to update the changelog

**Review by dgearhart — APPROVED** [review]
:shipit:

**Pull request merged**